exasol/python_extension_common/cli/std_options.py

```diff
@@ -0,0 +1,238 @@
+from typing import Any, no_type_check
+import os
+import re
+from enum import Flag, Enum, auto
+import click
+
+
+class ParameterFormatters:
+    """
+    Class facilitating customization of the cli.
+
+    The idea is that some of the cli parameters can be programmatically customized based
+    on values of other parameters and externally supplied formatters. For example a specialized
+    version of the cli may want to provide its own url. Furthermore, this url will depend on
+    the user supplied parameter called "version". The solution is to set a formatter for the
+    url, for instance "http://my_stuff/{version}/my_data". If the user specifies non-empty version
+    parameter the url will be fully formed.
+
+    A formatter may include more than one parameter. In the previous example the url could,
+    for instance, also include a username: "http://my_stuff/{version}/{user}/my_data".
+
+    Note that customized parameters can only be updated in a callback function. There is no
+    way to inject them directly into the cli. Also, the current implementation doesn't perform
+    the update if the value of the parameter dressed with the callback is None.
+    """
+    def __init__(self):
+        self._formatters = {}
```

Comment:
Add a type hint,probaly
```suggestion
        self._formatters : Dict[str, str] = {}
```